File: csharp/MapleStory.OpenAPI/Src/Dto/History/PotentialResultOptionDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 잠재능력 재설정 결과 옵션 정보
    /// </summary>
    public class PotentialResultOptionDTO
    {
        /// <summary>
        /// 옵션 명
        /// </summary>
        [JsonProperty("value")]
        public string Value { get; set; }

        /// <summary>
        /// 옵션 등급
        /// </summary>
        [JsonProperty("grade")]
        public string Grade { get; set; }

        public PotentialOptionGrade GradeEnum
        {
            get {
                return Utils.PotentialOptionGradeFromString(this.Grade);
            }
        }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/kms/dto/history/PotentialResultOptionDTO.java
package dev.spiralmoon.maplestory.api.kms.dto.history;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.spiralmoon.maplestory.api.common.dto.PotentialOptionGrade;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * 잠재능력 재설정 결과 옵션 정보
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class PotentialResultOptionDTO {

    /**
     * 옵션 명
     */
    @JsonProperty("value")
    private String value;

    /**
     * 옵션 등급
     */
    @JsonProperty("grade")
    private String grade;

    public PotentialOptionGrade getGradeEnum() {
        return PotentialOptionGrade.fromString(this.grade);
    }
}

